fla/layers/mom_varlen.py

```
@@ -0,0 +1,724 @@
+                cu_seqlens=cu_seqlen,
+                head_first=False
+            )
+            recurrent_state[0] = recurrent_state_
+            o_ = o_.squeeze(0).contiguous()
+            o_list = pad_input(o_, indices_q, batch_size*self.num_memories, max_len)
+            o_list = rearrange(o_list, '(e b) l h d -> e b l h d',b=batch_size)
+            o = reconstruct(o_list, indices=indices, sorted_indices=sorted_indices, batch_size=batch_size, seq_len=seq_len, topk=self.topk, routing_weights=routing_weights, mask=mask)
+
+        if self.shared_mem:
+            shared_o = self.shared_o(shared_hidden_states, attention_mask, recurrent_state, use_cache, conv_state_q, conv_state_k, conv_state_v)
+            o += shared_o
```

Pylint (3.3.7)

[error] Undefined variable 'o'

(E0602)